config/environments/test.rb

```diff
@@ -78,4 +78,6 @@
   config.after_initialize do
     PaperTrail.enabled = ENV['PAPERTRAIL_ENABLED'] || false
   end
+
+  config.cache_store = :memory_store
```

Comment:
I know very little about Ruby/Rails and even less about the rack-attack gem, but I'm guessing this line means it is storing attempts in local RAM. As we run multiple instances of `check-api` in production, each will have their own RAM. This means that the limits set in `Rack::Attack` will not be strictly accurate. E.g., we want no more than 5 login attemps from a single ip in 60 seconds, but if we are running 3 instances of `check-api` an attacker might get up to 3 x 5 = 15 attempts in 60 seconds before being throttled. Is there an option to store in Redis? I'm not sure what impact that could have on performance, however.

Reply:
I guess this line is actually about the test environment, where RAM is fine. I'm not sure how it's configured for QA/Live.

Reply:
Yes this is only on the test environment, but it will probably go away. Currently the rate limiting is not working on test, and I am trying to figure out why. Something seems to be disabling Rack::Attack. This line here was just one of the attempts to get it to work.